Configuration not passed to plugin

The tile pom (tile.pom.xml):

```
<project>
  <modelVersion>4.0.0</modelVersion>
  <groupId>it.session.maven.tiles</groupId>
  <artifactId>antrun1-tile</artifactId>
  <version>0.8-SNAPSHOT</version>
  <packaging>pom</packaging>

  <build>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-antrun-plugin</artifactId>
        <version>1.7</version>
        <executions>

         <execution>
            <id>print-antrun1</id>
            <phase>generate-sources</phase>
            <goals>
              <goal>run</goal>
            </goals>
          </execution>

          <execution>
            <id>print-antrun2</id>
            <phase>compile</phase>
            <configuration>
              <target>
                <echo message="compile execution config"/>
              </target>
            </configuration>
            <goals>
              <goal>run</goal>
            </goals>
          </execution>

        </executions>
        <configuration>
            <target>
                <echo message="plugin config"/>
            </target>
        </configuration>
      </plugin>
    </plugins>
  </build>
</project>
```

The pom using the tile (use.pom.xml):

```
<project>
    <modelVersion>4.0.0</modelVersion>
    <groupId>it.session.maven.tiles</groupId>
    <artifactId>antrun1-use</artifactId>
    <version>0.8-SNAPSHOT</version>

    <properties>
        <tile.1>it.session.maven.tiles:antrun1-tile:0.8-SNAPSHOT</tile.1>
    </properties>

    <build>
        <extensions>
            <extension>
                <groupId>it.session.maven.plugins</groupId>
                <artifactId>tiles-maven-plugin</artifactId>
                <version>0.8-beta-7-SNAPSHOT</version>
            </extension>
        </extensions>
    </build>

</project>
```

Output of `mvn -f tile.pom.xml install` as expected. Both echo's executed:

```
[INFO] ------------------------------------------------------------------------
[INFO] Building antrun1-tile 0.8-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-antrun-plugin:1.7:run (print-antrun1) @ antrun1-tile ---
[INFO] Executing tasks

main:
     [echo] plugin config
[INFO] Executed tasks
[INFO] 
[INFO] --- maven-antrun-plugin:1.7:run (print-antrun2) @ antrun1-tile ---
[INFO] Executing tasks

main:
     [echo] compile execution config
[INFO] Executed tasks
[INFO] 
[INFO] --- maven-install-plugin:2.3:install (default-install) @ antrun1-tile ---
[INFO] Installing ....
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

Output of `mvn -f use.pom.xml install` is missing echo from plugin configuration:

```
[INFO] Scanning for projects...
[INFO] Loaded Maven Tile 'it.session.maven.tiles:antrun1-tile:0.8-SNAPSHOT'
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building antrun1-use 0.8-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-antrun-plugin:1.7:run (print-antrun1) @ antrun1-use ---
[INFO] No ant target defined - SKIPPED
[INFO] 
[INFO] --- maven-resources-plugin:2.3:resources (default-resources) @ antrun1-use ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory .../src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.0.2:compile (default-compile) @ antrun1-use ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-antrun-plugin:1.7:run (print-antrun2) @ antrun1-use ---
[INFO] Executing tasks

main:
     [echo] compile execution config
[INFO] Executed tasks
[INFO] 
[INFO] --- maven-resources-plugin:2.3:testResources (default-testResources) @ antrun1-use ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory ..../src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.0.2:testCompile (default-testCompile) @ antrun1-use ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.10:test (default-test) @ antrun1-use ---
[INFO] No tests to run.
[INFO] Surefire report directory: ..../target/surefire-reports
```

Only the configuration from execution is passed to the antrun plugin. The plugin configuration is not.
What's strange is that the effective pom does contain the plugin configuraion:

```
...
<plugin>
  <artifactId>maven-antrun-plugin</artifactId>
  <version>1.7</version>
  <executions>
    <execution>
      <id>print-antrun1</id>
      <phase>generate-sources</phase>
      <goals>
        <goal>run</goal>
      </goals>
    </execution>
    <execution>
      <id>print-antrun2</id>
      <phase>compile</phase>
      <goals>
        <goal>run</goal>
      </goals>
      <configuration>
        <target>
          <echo message="compile execution config" />
        </target>
      </configuration>
    </execution>
  </executions>
  <configuration>
    <target>
      <echo message="plugin config" />
    </target>
  </configuration>
</plugin>
...
```

Another test case:
Tile pom (war-tile-pom.xml):

```
<project>
    <modelVersion>4.0.0</modelVersion>
    <groupId>it.session.maven.tiles</groupId>
    <artifactId>war-tile</artifactId>
    <version>0.8-SNAPSHOT</version>
    <packaging>pom</packaging>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-war-plugin</artifactId>
                <version>2.4</version>
                <configuration>
                    <warName>XXXXX</warName>
                    <failOnMissingWebXml>false</failOnMissingWebXml>
                </configuration>
            </plugin>
        </plugins>
    </build>
</project>
```

Pom using it (use-war.pom.xml):

```
<project>
    <modelVersion>4.0.0</modelVersion>
    <groupId>it.session.maven.tiles</groupId>
    <artifactId>war-use</artifactId>
    <version>0.8-SNAPSHOT</version>
    <packaging>war</packaging>

    <properties>
        <tile.1>it.session.maven.tiles:war-tile:0.8-SNAPSHOT</tile.1>
    </properties>

    <build>
        <extensions>
            <extension>
                <groupId>it.session.maven.plugins</groupId>
                <artifactId>tiles-maven-plugin</artifactId>
                <version>0.8-beta-7-SNAPSHOT</version>
            </extension>
        </extensions>
    </build>

</project>
```

Executing `mvn install -f ./war-tile-pom.xml && mvn install -f ./use-war.pom.xml` results in:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war 
(default-war) on project war-use: Error assembling WAR: webxml attribute is required 
(or pre-existing WEB-INF/web.xml if executing in update mode) -> [Help 1]
```

So `failOnMissingWebXml` is not passed to the war plugin

--- a/tiles-maven-plugin/src/main/java/it/session/maven/plugin/TilesModelMerger.java
+++ b/tiles-maven-plugin/src/main/java/it/session/maven/plugin/TilesModelMerger.java
@@ -1,11 +1,14 @@
 package it.session.maven.plugin;
 
-import org.apache.maven.model.Model;
-import org.apache.maven.model.Plugin;
-import org.apache.maven.model.merge.ModelMerger;
-
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.PluginExecution;
+import org.apache.maven.model.merge.ModelMerger;
 
 /**
  * TilesModelMerger by-passes the invocation to ModelMerger.merge() by adding the merge of Plugin configuration.
@@ -27,6 +30,13 @@ public class TilesModelMerger extends ModelMerger {
       for(Plugin sourcePlugin : source.getBuild().getPlugins()) {
         Plugin targetPlugin = target.getBuild().getPluginsAsMap().get(sourcePlugin.getKey());
         super.mergePlugin(targetPlugin, sourcePlugin, sourceDominant, context);
+        Set<Entry<String, PluginExecution>> entrySet = targetPlugin.getExecutionsAsMap().entrySet();
+        for (Entry<String, PluginExecution> entry : entrySet) {
+          PluginExecution execution = entry.getValue();
+          if (execution.getConfiguration() == null) {
+            execution.setConfiguration(sourcePlugin.getConfiguration());
+          }
+        }
       }
     }
   }

